ext4/tar2ext4/tar2ext4.go

```diff
@@ -272,10 +715,53 @@ func ConvertAndComputeRootDigest(r io.Reader) (string, error) {
 }
 
 // ConvertToVhd converts given io.WriteSeeker to VHD, by appending the VHD footer with a fixed size.
-func ConvertToVhd(w io.WriteSeeker) error {
+func ConvertToVhd(w io.WriteSeeker, alignToMB bool) error {
 	size, err := w.Seek(0, io.SeekEnd)
 	if err != nil {
 		return err
 	}
-	return binary.Write(w, binary.BigEndian, makeFixedVHDFooter(size))
+
+	diskSize := size
+	if alignToMB {
+		// diskPlusFooter is the disk size plus the size of the vhd footer to be added
+		diskPlusFooter := size + int64(vhdFooterSize)
+		if diskPlusFooter%bytesInMB != 0 {
+			remainder := bytesInMB - (diskPlusFooter % bytesInMB)
+			// seek the number of zeros needed to make this MB aligned after the vhd
+			// footer is added
+			diskSize, err = w.Seek(remainder, io.SeekCurrent)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return binary.Write(w, binary.BigEndian, makeFixedVHDFooter(diskSize))
+}
+
+func ReadVHDFooter(r io.ReadSeeker) (*VHDFooter, error) {
```

Comment:
For helper functions like this that are only used in the tool, I'd recommend defining them in the tool code itself, rather than here. I prefer to be conservative with adding new exported functions from public packages. We can always choose to export it later, but it's harder to go backwards once it's exported.

Reply:
I'll be removing this function all together since we're no longer going to export the "vhdFooter" struct